File: config/rules/opengrep/python/lang/cwe-798-hardcoded-credentials/hardcoded-credentials.py
import os

# ruleid: python.lang.security.cwe-798.hardcoded-credentials
password = "SuperSecret123!"

# ruleid: python.lang.security.cwe-798.hardcoded-credentials
API_KEY = "sk-proj-abc123def456ghi789"

# ruleid: python.lang.security.cwe-798.hardcoded-credentials
SECRET_KEY = "my-secret-key-do-not-share"

# ok: python.lang.security.cwe-798.hardcoded-credentials
password = os.environ.get("DB_PASSWORD")

# ok: python.lang.security.cwe-798.hardcoded-credentials
api_key = os.environ["API_KEY"]

# ok: python.lang.security.cwe-798.hardcoded-credentials
password = ""  # empty string is a valid default

File: config/rules/opengrep/csharp/lang/cwe-798-hardcoded-credentials/hardcoded-credentials.cs
public class HardcodedCredentialExamples
{
    // ruleid: csharp.lang.security.cwe-798.hardcoded-credentials
    private string ApiKey = "sk-proj-abc123def456";

    // ruleid: csharp.lang.security.cwe-798.hardcoded-credentials
    private string ConnectionString = "Server=db;Database=App;Password=SuperSecret123!;";

    // ok: csharp.lang.security.cwe-798.hardcoded-credentials
    public string GetApiKey(IConfiguration config)
    {
        return config["ApiKey"];
    }

    // ok: csharp.lang.security.cwe-798.hardcoded-credentials
    public string GetPassword()
    {
        return Environment.GetEnvironmentVariable("DB_PASSWORD");
    }
}
